### PR TITLE
Added better error message for handling incorrect payload

### PR DESCRIPTION
**Pull request by sunilk747**
Fix for issue - https://github.com/openshiftio/openshift.io/issues/4071

### CODE REVIEW

**Comment by sunilk747** [comment]
[test]

File: src/auth.py [hunk 1]
```diff
@@ -25,7 +26,7 @@ def get_audiences():
 def decode_token(token):
     """Decode JWT token entered by the user."""
     if token is None:
-        return {}
+        return flask.jsonify(dict(error='Auth token audience cannot be verified.')), 401
```
Comment (tisnik) [inline src/auth.py]: do you prefer `dict(x="y")` over `{"x": "y"}`?

File: src/auth.py [hunk 2]
```diff
@@ -74,26 +75,28 @@ def wrapper(*args, **kwargs):
...
+                return flask.jsonify(dict(error='Authentication failed - Not a valid Auth token '
+                                                'provided')), 401
 
             lgr.info('Successfuly authenticated user {e} using JWT'.
                      format(e=decoded.get('email')))
         except jwt.ExpiredSignatureError as exc:
             lgr.exception('Expired JWT token')
             decoded = {'email': 'unauthenticated@jwt.failed'}
-            raise HTTPError(401, 'Authentication failed - token has expired') from exc
+            return flask.jsonify('Authentication failed - token has expired'), 401
```
Comment (tisnik) [inline src/auth.py]: not a `dict()` here?

File: src/auth.py (hunk 1 above)
Comment (sunilk747) [reply src/auth.py]: Both do the same thing. If you prefer `{"x": "y"}` this way I can change it.

**Pull request merged**